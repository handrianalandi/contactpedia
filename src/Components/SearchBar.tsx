import styled from '@emotion/styled'
import React from 'react'
const SearchBarComponent = styled.input`
    border: 1px solid #ccc;
    border-radius: 15px;
    padding: 10px;
    width: 100%;
    margin: 0 auto;
    display: block;
    box-sizing: border-box;

    &:focus{
        outline: none;
        border: 2px solid #06BA63;
    }
`
interface SearchBarProps {
    setSearchTerm: (searchTerm: any) => void;
}
export default function SearchBar({setSearchTerm}: SearchBarProps) {
  return (
    <SearchBarComponent
        type="text"
        placeholder="Find by Name or Phone Number"
        onChange={(e) => setSearchTerm(e.target.value)}
    />
  )
}
