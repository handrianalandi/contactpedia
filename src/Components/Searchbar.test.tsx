import {render,fireEvent,screen} from '@testing-library/react';
import SearchBar from './SearchBar';
describe('Searchbar', () => {
    const mockOnChange = jest.fn();
    it('should render correctly', () => {
        const {container} = render(<SearchBar setSearchTerm={mockOnChange}/>);
        expect(container).toMatchSnapshot();
    }
    );

    it('should call onChange when input is changed', () => {
        render(<SearchBar setSearchTerm={mockOnChange}/>);
        //get the input
        const input = screen.getByTestId('search-bar');
        //fire the change event
        fireEvent.change(input, {target: {value: 'test'}});
        expect(mockOnChange).toHaveBeenCalledWith('test');
    }
    );
});