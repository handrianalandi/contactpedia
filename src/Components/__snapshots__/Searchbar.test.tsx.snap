// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar should render correctly with props 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

.emotion-1 {
  height: 1.3rem;
  width: 1.3rem;
  position: absolute;
  left: 0.5rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  color: #656565;
}

.emotion-2 {
  border: 1px solid #ccc;
  border-radius: 15px;
  padding: 10px;
  width: 100%;
  margin: 0 auto;
  display: block;
  box-sizing: border-box;
  width: 100%;
  padding-left: 2.5rem;
}

.emotion-2:focus {
  outline: none;
  border: 2px solid #06ba63;
}

<div>
  <div
    class="emotion-0"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-magnifying-glass  emotion-1"
      data-icon="magnifying-glass"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
        fill="currentColor"
      />
    </svg>
    <input
      class="emotion-2"
      data-testid="search-bar"
      placeholder="Test Placeholder"
      type="text"
    />
  </div>
</div>
`;
